addons/mail/models/mail_channel.py

```diff
@@ -1153,6 +1158,28 @@ def channel_create(self, name, privacy='public'):
         self.env['bus.bus'].sendone((self._cr.dbname, 'res.partner', self.env.user.partner_id.id), channel_info)
         return channel_info
 
+    @api.model
+    def group_chat_create(self, partners_to):
+        """ Create a private group chat with at least 3 partners.
+            :param partners_to : list of res.partner ids to add to the conversation
+            :returns: channel_info of the created or existing channel
+            :rtype: dict
+        """
+        if self.env.user.partner_id.id not in partners_to:
+            partners_to.append(self.env.user.partner_id.id)
+        channel = self.create({
+            'channel_partner_ids': [Command.link(partner_id) for partner_id in partners_to],
+            'public': 'private',
+            'channel_type': 'group',
+            'email_send': False,
+            'name': ', '.join(self.env['res.partner'].sudo().browse(partners_to).mapped('name')),
```

Comment:
Prefer using `search` instead of `browse`, as `browse` would unexpectedly crash if given invalid ids.
Also if you need to get recordset of partners anyway, it would be better to do it once at the top of the method, and use the recodset everywhere instead of the `partners_to` param.